File: HetDemo/app/src/main/java/com/junkchen/hetdemo/MainActivity.java
package com.junkchen.hetdemo;

import android.content.Intent;
import android.graphics.Color;
import android.graphics.Typeface;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.text.Spannable;
import android.text.SpannableString;
import android.text.style.AbsoluteSizeSpan;
import android.text.style.ForegroundColorSpan;
import android.text.style.RelativeSizeSpan;
import android.text.style.StyleSpan;
import android.util.DisplayMetrics;
import android.util.TypedValue;
import android.view.View;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        DisplayMetrics metrics = new DisplayMetrics();
        this.getWindowManager().getDefaultDisplay().getMetrics(metrics);

        TextView tv_hello = findViewById(R.id.tv_hello);
        String hours = "1";
        int hoursLen = hours.length();
        String minutes = "25";
        int minutesLen = minutes.length();
        String text = "1小时25分钟";
        SpannableString spannableString = new SpannableString(text);
        spannableString.setSpan(new ForegroundColorSpan(Color.BLACK),
                0, hoursLen, Spannable.SPAN_EXCLUSIVE_INCLUSIVE);
        spannableString.setSpan(new ForegroundColorSpan(Color.BLACK),
                hoursLen + 2, hoursLen + 2 + minutesLen, Spannable.SPAN_EXCLUSIVE_INCLUSIVE);
        spannableString.setSpan(new StyleSpan(Typeface.BOLD),
                0, hoursLen, Spannable.SPAN_EXCLUSIVE_INCLUSIVE);
        spannableString.setSpan(new StyleSpan(Typeface.BOLD),
                hoursLen + 2, hoursLen + 2 + minutesLen, Spannable.SPAN_EXCLUSIVE_INCLUSIVE);
        spannableString.setSpan(new AbsoluteSizeSpan(
                        (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_SP, 20, metrics)),
                0, hoursLen, Spannable.SPAN_EXCLUSIVE_INCLUSIVE);
        spannableString.setSpan(new RelativeSizeSpan(1.66F),
                hoursLen + 2, hoursLen + 2 + minutesLen, Spannable.SPAN_EXCLUSIVE_INCLUSIVE);
        tv_hello.setText(spannableString);

    }

    public void doClick(View view) {
        if (view.getId() == R.id.tv_hello) {
            startActivity(new Intent(this, SurveyActivity.class));
        } else {
            startActivity(new Intent(this, AnswerActivity.class));
        }
    }
}

File: LKDemo/viewstubdemo/src/main/java/com/junkchen/viewstubdemo/MainActivity.kt
package com.junkchen.viewstubdemo

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.view.ViewStub
import android.widget.TextView

class MainActivity : AppCompatActivity() {

//    val TAG = MainActivity::class.simpleName

    lateinit var mViewStub: ViewStub
    lateinit var txtv_content: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        mViewStub = findViewById(R.id.stub)
    }

    fun doClick(view: View) {
        when (view.id) {
            R.id.btn_inflate -> {
                val view = mViewStub.inflate()
                txtv_content = findViewById(R.id.txtv_content)
            }
            R.id.btn_set -> {
//                txtv_content.setText("I'm new data.")
                txtv_content.text = "I'm new data."
            }
        }
    }
}
